File: retropreference/src/main/java/com/twiceyuan/library/typeHandler/IntegerHandler.java
package com.twiceyuan.library.typeHandler;

import android.content.SharedPreferences;

/**
 * Created by twiceYuan on 20/01/2017.
 *
 * Integer handler
 */
public class IntegerHandler extends BaseTypeHandler<Integer> {

    public IntegerHandler(SharedPreferences preferences) {
        super(preferences);
    }

    @Override
    public void setValue(String key, Integer value) {
        mPreferences.edit().putInt(key, value).apply();
    }

    @Override
    public Integer get(String key, Integer defaultValue) {
        return mPreferences.getInt(key, defaultValue);
    }

    @Override
    public Integer defaultValue() {
        return 0;
    }
}

File: retropreference/src/main/java/com/twiceyuan/retropreference/typeHandler/IntegerHandler.kt
package com.twiceyuan.retropreference.typeHandler

import android.content.SharedPreferences

/**
 * Created by twiceYuan on 20/01/2017.
 * 
 * Integer handler
 */
internal class IntegerHandler(preferences: SharedPreferences) : BaseTypeHandler<Int>(preferences) {

    override fun setValue(key: String, value: Int?) {
        mPreferences.edit().putInt(key, value ?: 0).apply()
    }

    override operator fun get(key: String, defaultValue: Int?): Int? = mPreferences.getInt(key, defaultValue ?: 0)

    override fun defaultValue(): Int? = 0
}
